src/librustdoc/html/static/js/externs.js

```diff
@@ -8,10 +8,30 @@ function initSearch(searchIndex){}
 
 /**
  * @typedef {{
- *   raw: string,
- *   query: string,
- *   type: string,
- *   id: string,
+ *     isExact: boolean,
```

Comment:
We decided that everything would be an exact match except for a top-level "simple" query. So it doesn't make sense to attach `isExact` to each QueryElement. Instead we should have something like: at the top level, a `ParsedQuery` contains: `parsed: (BasicQuery|ExactQuery|TypedQuery)`. And then when we are processing a TypedQuery (e.g. `Foo<Bar>`), we assume we are always doing an exact match.